### PR TITLE
feat(NODE-5008): add zstd and kerberos to peer deps

### PR DESCRIPTION
**Pull request by durran**
### Description

Adds zstd and kerberos to peer dependencies.

#### What is changing?

Sets @mongodb-js/zstd and kerberos in the peer dependencies to their latest versions and sets them as optional.

##### Is there new documentation needed for these changes?

None

#### What is the motivation for this change?

NODE-5008

### Double check the following

- [x] Ran `npm run check:lint` script
- [x] Self-review completed using the [steps outlined here](https://github.com/mongodb/node-mongodb-native/blob/HEAD/CONTRIBUTING.md#reviewer-guidelines)
- [x] PR title follows the [correct format](https://www.conventionalcommits.org/en/v1.0.0/): `type(NODE-xxxx)[!]: description`
  - Example: `feat(NODE-1234)!: rewriting everything in coffeescript`
- [] Changes are covered by tests
- [ ] New TODOs have a related JIRA ticket

### CODE REVIEW

File: package.json
Comment (baileympearson) [inline package.json:38]: Unless we plan on leaving this PR open until we begin 6.0 work, we need to include `^1.0.0` as well in the semver for kerberos.

**Review by baileympearson — CHANGES_REQUESTED** [review]
requesting changes for visibilty

File: package.json
Comment (durran) [reply package.json:38]: Just waiting on team opinion on whether this can go in as is or not. I believe 6.0 is starting now.

Comment (durran) [reply package.json]: Updated to `1.0.0 || ^2.0.0`

**Review by baileympearson — APPROVED** [review]
waiting on CI, LGTM

**Pull request merged**
